			     BASH PATCH REPORT
			     =================

Bash-Release:	4.4
Patch-ID:	bash44-018

Bug-Reported-by:	Siteshwar Vashisht <svashisht@redhat.com>
Bug-Reference-ID:	<1341922391.30876471.1501250355579.JavaMail.zimbra@redhat.com>
Bug-Reference-URL:	https://bugzilla.redhat.com/show_bug.cgi?id=1466737

Bug-Description:

Under certain circumstances (e.g., reading from /dev/zero), read(2) will not
return -1 even when interrupted by a signal. The read builtin needs to check
for signals in this case.

Patch (apply with `patch -p0'):

*** ../bash-20170622/builtins/read.def	2017-06-17 18:45:20.000000000 -0400
--- builtins/read.def	2017-06-30 11:09:26.000000000 -0400
***************
*** 611,615 ****
  
        CHECK_ALRM;
! 
  #if defined (READLINE)
  	}
--- 611,615 ----
  
        CHECK_ALRM;
!       QUIT;		/* in case we didn't call check_signals() */
  #if defined (READLINE)
  	}
*** ../bash-4.4/patchlevel.h	2016-06-22 14:51:03.000000000 -0400
--- patchlevel.h	2016-10-01 11:01:28.000000000 -0400
***************
*** 26,30 ****
     looks for to find the patch level (for the sccs version string). */
  
! #define PATCHLEVEL 17
  
  #endif /* _PATCHLEVEL_H_ */
--- 26,30 ----
     looks for to find the patch level (for the sccs version string). */
  
! #define PATCHLEVEL 18
  
  #endif /* _PATCHLEVEL_H_ */
